pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -742,6 +742,8 @@ def set_distributed_mode(self, distributed_backend: Optional[str] = None):
 
         # special case with DDP on CPUs
         if self.distributed_backend == "ddp_cpu":
+            if _TPU_AVAILABLE:
+                raise MisconfigurationException("`accelerator='ddp_cpu'` is not supported on TPU machines")
```

Comment:
So if the machine has a TPU at all, ddp cpu does not work? Even if they are disabled/unused?

Reply:
No idea. Even a few folks at XLA team didn't have an exact solution. You could read more about it in #7810